ENIGMAsystem/SHELL/Bridges/General/Win32-Direct3D9.cpp

```diff
@@ -105,18 +111,71 @@ namespace enigma
     if (forceSoftwareVertexProcessing) {
       behaviors = D3DCREATE_SOFTWARE_VERTEXPROCESSING;
     }
-    hr = d3dobj->CreateDevice(D3DADAPTER_DEFAULT,
-                              D3DDEVTYPE_HAL,
-                              hWnd,
-                              behaviors,
-                              &d3dpp,
-                              &d3dmgr->device);
-    if (FAILED(hr)) {
-      MessageBox(hWnd,
-                 "Failed to create Direct3D 9.0 Device",
-                 DXGetErrorDescription9(hr), //DXGetErrorString9(hr)
-                 MB_ICONERROR | MB_OK);
-      return; // should probably force the game closed
+
+    // Manually load the d3d9.dll library to check for D3D9Ex
+    HMODULE libHandle = NULL;
+    libHandle = LoadLibrary("d3d9.dll");
```

Comment:
Is this seriously how MSDN recommends using this method? Or are you working around MinGW inadequacies?

Reply:
Yes, that's seriously how MSDN recommends checking for this. Please see the second example at the link below.
https://docs.microsoft.com/en-us/windows/desktop/api/d3d9/nf-d3d9-direct3dcreate9ex#examples

I should clarify this to you too:
Understand that because we don't copy the Direct3D 9 dll next to the compiled game like GMSv1.4, we require installation of the DirectX 9.0 End User Runtime. If the user doesn't have the runtime installed, then the dll does not exist.
![GMSv1.4 Copies D3D9 DLL](https://user-images.githubusercontent.com/3212801/50063001-38abd780-0178-11e9-9589-5498445e2a78.png)

Reply:
That's incredibly sad, but all right. Fire away.